### PR TITLE
add script to delete unattached GCP "Sourcegraph Auxiliary" disks 

### PR DESCRIPTION
**Pull request by ggilmore**
- We have a limited disk quota on the Sourcegraph Auxiliary project, so unused disks should be removed promptly 

- Nobody should be creating long-lived services in the Sourcegraph Auxiliary project, so it's safe to delete these disks. (You should create a snapshot if you want to retain data after you delete an instance)

- This script should ultimately run as a cronjob on our CI cluster, but that requires some additional configuration. Adding it as a simple script in this repo is faster, and is still useful since this repository's integration tests create many of these disks.

### CODE REVIEW

File: .buildkite/cleanup-unused-disks.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+PROJECT=sourcegraph-server
+
+gcloud_command() {
+    gcloud --quiet --project="$PROJECT" "$@"
+}
+
+echo "--- Deleting unattached GCP disks from the $PROJECT project"
+
+for disk in $(gcloud_command compute disks list --filter="-users:*" --format="value(selfLink)")
```
Comment (beyang) [inline .buildkite/cleanup-unused-disks.sh]: Does this *only* match unattached disks? When I ran this against the `sourcegraph-dev` project, it included disks like `pgsql-prod-dot-com` and `zoekt-data`, which seem very much attached and critical.

Comment (ggilmore) [reply .buildkite/cleanup-unused-disks.sh]: Screen Shot 2019-10-16 at 4 17 56 PM
@beyang Those disks that you mentioned are indeed unattached.

Comment (ggilmore) [reply .buildkite/cleanup-unused-disks.sh]: @beyang Additionally, ["Disks can be deleted only if they are not being used by any virtual machine instances"](https://cloud.google.com/sdk/gcloud/reference/compute/disks/delete).

Comment (beyang) [reply .buildkite/cleanup-unused-disks.sh]: worth calling out in a comment, I think. Otherwise, LGTM.

**Pull request merged**